File: python/bob/bob.py
def hey(what):
    what = what.strip()
    if not what:
        return 'Fine. Be that way!'
    elif what.upper() == what and what.upper() != what.lower():
        return 'Whoa, chill out!'
    elif what[-1] == '?':
        return "Sure."
    else:
        return 'Whatever.'

File: ruby/bob/bob.rb
# Bob is a lackadaisical teenager. His responses are very limited.
class Bob
  RESPONSES = ['Sure.',
               'Whoa, chill out!',
               'Fine. Be that way!',
               'Whatever.'].freeze

  def self.hey(remark)
    stripped_remark = remark.strip
    if stripped_remark == stripped_remark.upcase &&
       stripped_remark.upcase != stripped_remark.downcase # checks for letters
      return RESPONSES[1]
    elsif stripped_remark.end_with?('?')
      return RESPONSES[0]
    elsif stripped_remark == ''
      return RESPONSES[2]
    end
    RESPONSES[3]
  end
end

module BookKeeping
  VERSION = 1
end
